apis/apiextensions/v1/xrd_types.go

```diff
@@ -77,10 +77,12 @@ type CompositeResourceDefinitionSpec struct {
 	// are sorted first by GA > beta > alpha (where GA is a version with no
 	// suffix such as beta or alpha), and then by comparing major version, then
 	// minor version. An example sorted list of versions: v10, v2, v1, v11beta2,
-	// v10beta3, v3beta1, v12alpha1, v11alpha2, foo1, foo10. Note that all
-	// versions must have identical schemas; Crossplane does not currently
-	// support conversion between different version schemas.
+	// v10beta3, v3beta1, v12alpha1, v11alpha2, foo1, foo10.
 	Versions []CompositeResourceDefinitionVersion `json:"versions"`
+
+	// Conversion defines all conversion settings for the defined Composite resource.
+	// +optional
+	Conversion *extv1.CustomResourceConversion `json:"conversion,omitempty"`
```

Comment:
Should we use the `extv1` type here, or define our own?

I'm thinking we may want to support strategies that upstream doesn't support in future, for example some kind of declarative conversion. I suppose we could always switch from upstream's type to our own at that point.

Reply:
Yes, I also had the same doubt, in the end I chose `extv1` because I thought it was better to effortlessly support everything they might introduce upstream, and as you said, we can always switch to our own later on if needed. But I'm ok switching to our own if you feel we want to have more control.

Reply:
> we can always switch to our own later on if needed. 

@phisco can you validate that this is possible without changing (i.e. breaking) the API? If possible, we can go with `extv1` for now and consider this later.